### ISSUE TITLE
Improve desktop encrypted storage

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/uniffi-rs#624]
**TL;DR Why:** Build a encrypted database that stores the encryption keys for accessing passwords on desktop. Currently we use Master Password to store the encryption keys, however, this solution is not as secure as we'd like it to be, additionally, it's not a maintained feature.

On mobile, we currently store most of the profile un-encrypted  and just encrypt keys for passwords, and this is all stored in the OS keychain to secure the data. However, on desktop we cannot use the same mechanism due to technical challenges. Therefore, we need to make an encrypted storage database that can store the encrypted keys for passwords. Right now, the encrypted keys can be accessed through the users Master Password, but we'd like to move away from this solution as it's not maintained or the most secure solution.

The logins team would be a primary consumer of this work, but the Web Platform team is also looking into trying to create an encrypted storage database for their own use case around private browsing.

**Comment by data-sync-user** [comment]
➤ Ana Medinac commented:

Later. Dependent on external teams and decisions around Master Password, and crypto work.

**Comment by data-sync-user** [comment]
➤ Ana Medinac commented:

Later. This work is still waiting on a few external decisions to be made around Master password and crypto work. This project will likely be revisted in late 2020, once we evaluate demand and once we've completed the FxA integration.
